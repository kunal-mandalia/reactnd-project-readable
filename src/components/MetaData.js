import React from 'react'
import PropTypes from 'prop-types'
import moment from 'moment'
import '../styles/MetaData.css'

const MetaData = ({ date, author }) => {
  const daysSince = moment().diff(date, 'days')
  const minutesSince = moment().diff(date, 'days')
  return (
    <div className='meta-data'>
      <div className='date'>
        {daysSince > 0 ? `${daysSince}d` : `${minutesSince}m`}
      </div>
      <div className='author'>
        {author}
      </div>
    </div>
  )
}

MetaData.propTypes = {
  date: PropTypes.number,
  author: PropTypes.string
}

export default MetaData
