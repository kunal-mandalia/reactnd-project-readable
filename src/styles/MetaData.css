.meta-data {
  display: inline-flex;
  font-size: small;
  color: lightgrey;
  margin-top: 4px;
}

.date {
  margin-right: 4px;
}
