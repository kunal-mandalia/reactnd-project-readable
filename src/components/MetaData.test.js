import React from 'react'
import MetaData from './MetaData'
import renderer from 'react-test-renderer'

const props = {
  date: Date.now(),
  author: `Kunal Mandalia`
}

describe(`<MetaData .../>`, () => {
  it(`should match snapshot`, () => {
    const tree = renderer.create(<MetaData {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
