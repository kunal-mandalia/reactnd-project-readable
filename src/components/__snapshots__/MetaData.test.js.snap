// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MetaData .../> should match snapshot 1`] = `
<div
  className="meta-data"
>
  <hr />
  <div
    className="date"
  >
    0m
  </div>
  <div
    className="author"
  >
    Kunal Mandalia
  </div>
</div>
`;
